crema/parsers/comet.py

```diff
@@ -0,0 +1,209 @@
+
+    psms = read_txt(
+        data,
+        target_column=target,
+        spectrum_columns=spectrum,
+        score_columns=scores,
+        peptide_column=peptide,
+        protein_column=protein,
+        protein_delim=protein_delim,
+        sep="\t",
+        copy_data=False,
+    )
+
+    # Remove first and last amino acid from sequence
+    # Looks like "R.WVNEK.Y"
+    peptide_column = psms.peptides
+    new_peptide_column = peptide_column.str[2:-2]
+    psms.set_peptide_column(new_peptide_column)
+
+    # always pair target and decoys for Comet
+    if pairing_file_name == None:
+        # implicit pairing based off fact that Comet reverses peptides
+        psms._peptide_pairing = _create_pairing(
+            data, peptide, protein, crux_comet
+        )
+    else:  # explicit pairing
+        psms._peptide_pairing = utils.create_pairing_from_file(
+            pairing_file_name
+        )
+
+    # Remove decoy prefix from protein ID
+    protein_column = psms.proteins
+    new_protein_column = protein_column.str.replace("DECOY_", "", regex=True)
+    psms.set_protein_column(new_protein_column)
+
+    return psms
+
+
+def _create_pairing(pairing_data, peptide_col, protein_col, crux_comet):
+    """Parse a single Comet dataframe to implicity pair target and
+    decoy sequences.
+
+    Parameters
+    ----------
+    pairing_data : pandas.DataFrame
+        A collection of PSMs with the necessary columns to create a
+        target/decoy peptide pairing. Required columns are modified sequence
+        and protein ID.
+
+    peptide_col: str
+        Identifies the modified sequence column.
+
+    protein_col: str
+        Identifies the protein ID column.
+
+    crux_comet: bool
+        If true, then output is from Comet inside of Crux. If false, then outut
+        if from standalone Comet.
+
+    Returns
+    -------
+    pairing : dict
+        A map of target and decoy peptide sequence pairings. Targets with
+        missing decoys will not be included among the keys.
+
+    """
+    # ensure pairing_data dataframe contains all necessary columns
+    req_fields = [peptide_col, protein_col]
+
+    if not set(req_fields).issubset(pairing_data.columns):
+        miss = ", ".join(set(req_fields) - set(pairing_data.columns))
+        raise ValueError(
+            f"Required columns for peptide pairing were not detected: {miss}"
+        )
+
+    pairing_data = pairing_data.loc[:, req_fields]
+    pairing_data[peptide_col] = pairing_data[peptide_col].str[2:-2]
+
+    if crux_comet:
+        pairing_data["target/decoy"] = ~pairing_data[protein_col].str.contains(
+            "decoy_"
+        )
+    else:
+        pairing_data["target/decoy"] = ~pairing_data[protein_col].str.contains(
+            "DECOY_"
+        )
+
+    reverse_peptide_list = []
+    for seq in list(pairing_data[peptide_col]):
+        # NOTE regex is not quite right
+        seq_sp = re.split(r"([A-Z]\[\d+\.\d+\])?", seq)
```

Comment:
**todo:** What's wrong with this regex? We should fix whatever it is...

Reply:
The regex isn't wrong per say. The final result is correct (after next two lines), It just doesn't do exactly what I want

Let's say we have seq="QQMM[15.9949]TGNR". Once the split occurs seq_sp=['', None, 'Q', None, 'Q', None, 'M', 'M[15.9949]', '', None, 'T', None, 'G', None, 'N', None, 'R', None, '']

The next two lines remove "" and None so that I can reverse the peptide.

Does that make sense? I couldn't figure out the regex where you split a string by amino acid, except when a mod occurs.

Reply:
Does this regex, `(?<=.)(?=[A-Z])`, work more like you'd expect?

Reply:
Perfect, thanks!